disk.go

```diff
@@ -19,21 +20,65 @@ const (
 //
 //==================================================================
 type Disk struct {
-	node       RunnerInterface
-	MBps       int
-	reserved   int
-	lastIOdone time.Time
-	writes     int64
-	reads      int64
-	writebytes int64
-	readbytes  int64
+	node           RunnerInterface
+	MBps           int
+	reserved       int
+	lastIOdone     time.Time
+	writes         int64
+	reads          int64
+	writebytes     int64
+	readbytes      int64
+	stopHandler    bool
+	pendingwbytes  int64
+	pendingrbytes  int64
+	qMutex         sync.Mutex
 }
 
 func NewDisk(r RunnerInterface, mbps int) *Disk {
 	d := &Disk{node: r, MBps: mbps, reserved: 0, lastIOdone: Now}
 	return d
 }
 
+//
+// This method is to simulate any disk controller actions.
+// The minimal handler here simulates the writes. Wakes up
+// after every unit of time needed to write a chunk and then
+// updates the pendingwbytes which is used to derive queue depth.
+//
+func (d *Disk) handler() {
+	// time required to write one chunk.
```

Comment:
The runner has a lot of other interfaces that are not relevant to Disk. Probably we canrefactor runner to a more abstract base interface. Will try to refactor and see